vhdbuilder/scripts/automate_ev2pipeline_trigger.sh

```diff
@@ -5,17 +5,29 @@ source vhdbuilder/scripts/automate_helpers.sh
 
 echo "Triggering ev2 artifact pipeline with Build ID $1"
 
-build_id=$1
-
-set +x
-system_access_token=$2
+build_ids=$1
 
 trigger_pipeline() {
-    az pipelines variable update --name VHD_PIPELINE_RUN_ID --pipeline-id 165939 --value $build_id  # Update the VHD_PIPELINE_RUN_ID with the build ID
-    az pipelines run --id 165939  # 165939 is the pipeline ID for ev2 artifacts pipeline
-}
+    pipeline_id="165939"   # 165939 is the pipeline ID for ev2 artifacts pipeline
+    for build_id in $build_ids; do
+        echo "Build ID for the release is $build_id"
+        az pipelines variable update --name VHD_PIPELINE_RUN_ID --pipeline-id $pipeline_id --value $build_id  # Update the VHD_PIPELINE_RUN_ID with the build ID
+        # In case auth fails/other issue, we do not want the pipeline to run if the build ID was not correctly updated
+        if [[ $build_id != $(az pipelines variable list --pipeline-id $pipeline_id | jq -r '.VHD_PIPELINE_RUN_ID.value') ]]; then
+            echo "Build ID failed to update, cancel operation"
+            exit 1
+        else
+            echo "Build ID successfully updated"
+        fi
 
-configure_az_devops $system_access_token
-set -x
+        # Run the pipeline and fetch the run ID to poll for success later
+        run_id=$(az pipelines run --id $pipeline_id | jq -r '.id')
+        while ! az pipelines runs show --id $run_id | grep -q '"result": "succeeded"'; do
```

Comment:
The pipeline "result" is null until it succeeds. Should a "failed" state be considered? This logic presumes that the ev2 artifact pipeline will eventually succeed.

Reply:
I think as long as this has a reasonable timeout it's okay to just search for the success case

Reply:
So around a 2 hr timeout for the polling to detect success or exit the loop would be good? I've seen some builds take up to 1.5 hrs.

Reply:
This is ev2 artifact build, it doesn't take more than 3/4 minutes.

Reply:
Ok, got it - thought it was the build for some reason :). The ev2 build artifact runs have usually succeeded so a timeout a bit more generous than the usual time to complete would be good.